docs/guides/typescript.md

```diff
@@ -234,14 +234,17 @@ If you are eager to know what the answer is to this particular problem then you
 
 ### Handling Dynamic `replace` Flag
 
-If the value of the `replace` flag is not known at compile time and is determined dynamically, you might face issues. To handle this, you can use a workaround by annotating the `replace` parameter with `as any`:
+If the value of the `replace` flag is not known at compile time and is determined dynamically, you might face issues. To handle this, you can use a workaround by annotating the `replace` parameter with the parameters of the `setState` function:
 
 ```ts
 const replaceFlag = Math.random() > 0.5
-store.setState(partialOrFull, replaceFlag as any)
+const args = [{ bears: 5 }, replaceFlag] as Parameters<
+  typeof useBearStore.setState
+>
+store.setState(...args)
 ```
 
-#### Example with `as any` Workaround
```

Comment:
A working example can be found here: https://stackblitz.com/edit/vitejs-vite-wqmf9c?file=src%2FApp.tsx